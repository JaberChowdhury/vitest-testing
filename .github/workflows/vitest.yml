name: Run Vitest Tests

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      # Step 1: Check out the repository
      - name: Checkout code
        uses: actions/checkout@v3

      # Step 2: Set up Node.js with pnpm caching
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18
          cache: "pnpm"

      # Step 3: Install pnpm manually (bypassing Corepack)
      - name: Install pnpm
        run: npm install -g pnpm

      # Step 4: Ensure pnpm is added to PATH
      - name: Add pnpm to PATH
        run: echo "$(npm root -g)/pnpm" >> $GITHUB_PATH

      # Step 5: Verify pnpm installation and check PATH
      - name: Verify pnpm installation
        run: |
          echo "pnpm version:"
          pnpm --version
          echo "pnpm location:"
          which pnpm
          echo "Current PATH:"
          echo $PATH

      # Step 6: Install dependencies
      - name: Install dependencies
        run: pnpm install

      # Step 7: Run Vitest tests
      - name: Run tests
        run: pnpm test
