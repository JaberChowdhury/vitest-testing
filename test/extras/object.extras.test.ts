import { beforeEach, describe, expect, test, vi } from "vitest";

describe("Object extras", () => {
  //    Testing Objects
  const user = {
    name: "John Doe",
    age: 30,
    isAdmin: false,
  };

  test("user object should have correct properties", () => {
    expect(user).toEqual({
      name: "John Doe",
      age: 30,
      isAdmin: false,
    });
  });

  test("user should not be an admin", () => {
    expect(user.isAdmin).toBe(false);
  });
});
describe("Object extras", () => {
  // Testing Asynchronous Code

  function fetchData() {
    return new Promise((resolve) => {
      setTimeout(() => resolve("data"), 1000);
    });
  }

  test('fetchData should return "data"', async () => {
    await expect(fetchData()).resolves.toBe("data");
  });
});
describe("Object extras", () => {
  // async / await;

  async function fetchData() {
    return "data";
  }

  test('fetchData should return "data"', async () => {
    const data = await fetchData();
    expect(data).toBe("data");
  });
});
describe("Object extras", () => {
  // Mocking;

  const fetchData = vi.fn(() => Promise.resolve("data"));

  test('fetchData should return "data"', async () => {
    const result = await fetchData();
    expect(result).toBe("data");
    expect(fetchData).toHaveBeenCalled();
  });
});
describe("Object extras", () => {
  // Snapshot Testing

  function generateMarkup() {
    return "<div>Hello, World!</div>";
  }

  test("markup should match snapshot", () => {
    const markup = generateMarkup();
    expect(markup).toMatchSnapshot();
  });
});
describe("Object extras", () => {
  // Testing Edge Cases

  function divide(a, b) {
    if (b === 0) throw new Error("Division by zero");
    return a / b;
  }

  test("divide should throw an error when dividing by zero", () => {
    expect(() => divide(10, 0)).toThrow("Division by zero");
  });
});
describe("Object extras", () => {
  let counter = 0;

  beforeEach(() => {
    counter = 0;
  });

  test("increment counter", () => {
    counter++;
    expect(counter).toBe(1);
  });

  test("reset counter", () => {
    expect(counter).toBe(0);
  });
});
