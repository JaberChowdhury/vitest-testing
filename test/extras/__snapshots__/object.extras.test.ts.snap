// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Object extras > markup should match snapshot 1`] = `"<div>Hello, World!</div>"`;
